library/lib/components/CustomControls.tsx

```diff
@@ -1,12 +1,72 @@
 import { Controls, useReactFlow, useStore } from "@xyflow/react"
+import { useDiagramStore, useMetadataStore } from "@/store/context"
+import { useShallow } from "zustand/shallow"
+import { UndoIcon } from "./Icon/UndoIcon"
+import { RedoIcon } from "./Icon/RedoIcon"
+import { Tooltip } from "@mui/material"
+import { ApollonMode } from "@/typings"
 
 export const CustomControls = () => {
   const { zoomTo } = useReactFlow()
   const zoomLevel = useStore((state) => state.transform[2])
   const zoomLevelPercent = Math.round(zoomLevel * 100)
 
+  const mode = useMetadataStore(useShallow((state) => state.mode))
+
+  // Create condition for showing undo/redo buttons
+  const showUndoRedo = mode === ApollonMode.Modelling
```

Comment:
Instead of directly checking the condition or managing the undo/redo logic here, we simply check if the undoManager is initialized in the store. This way, if the initialization logic changes later, we don't need to update this part of the code.


```suggestion

  const { canUndo, canRedo, undo, redo } = useDiagramStore(
    useShallow((state) => ({
      canUndo: state.canUndo,
      canRedo: state.canRedo,
      undo: state.undo,
      redo: state.redo,
      undoManagerExist: state.undoManger !== null
    }))
  )
```